src/common/android/deviceResolver.ts

```diff
@@ -0,0 +1,37 @@
+import {ChildProcess} from "../node/childProcess";
+import * as Q from "q";
+
+export interface IDevice {
+    id: string;
+    status: string;
+}
+
+export class DeviceResolver {
+
+    /**
+     * Gets the list of Android connected devices and emulators.
+     */
+    public getConnectedDevices(): Q.Promise<IDevice[]> {
+        let deferred = Q.defer<IDevice[]>();
```

Comment:
I think you can remove this deferred  and just chain the results of exectToString